src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -30,29 +32,181 @@ const ActorInit Obj_Lift_InitVars = {
+
+extern Gfx D_06000140[]; //gCollapsingPlatformDL
+extern CollisionHeader D_060003D0; //gCollapsingPlatformCol
+
+void ObjLift_SetupAction(ObjLift* this, ObjLiftActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ObjLift_InitDynaPoly(ObjLift* this, GlobalContext* globalCtx, CollisionHeader* collision, DynaPolyMoveFlag flags) {
+    s32 pad;
+    CollisionHeader* colHeader = NULL;
+    s32 pad2;
+
+    DynaPolyActor_Init(&this->dyna, flags);
+    CollisionHeader_GetVirtual(collision, &colHeader);
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+    if (this->dyna.bgId == BG_ACTOR_MAX) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_obj_lift.c",
+                     188, this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_80B96160(ObjLift* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f* temp_s3;
+    s32 pad0;
+    s32 i;
+
+    temp_s3 = &this->dyna.actor.world.pos;
+    for (i = 0; i < ARRAY_COUNT(D_80B969D0); i++) {
+        pos.x = D_80B969D0[i].x * this->dyna.actor.scale.x + temp_s3->x;
+        pos.y = temp_s3->y;
+        pos.z = D_80B969D0[i].z * this->dyna.actor.scale.z + temp_s3->z;
+        velocity.x = D_80B969D0[i].x * this->dyna.actor.scale.x * 0.8f;
+        velocity.y = Rand_ZeroOne() * 10.0f + 6.0f;
+        velocity.z = D_80B969D0[i].z * this->dyna.actor.scale.z * 0.8f;
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, temp_s3, -0x100, (Rand_ZeroOne() < 0.5f) ? 0x40 : 0x20, 0xF,
+                             0xF, 0, (Rand_ZeroOne() * 50.0f + 50.0f) * this->dyna.actor.scale.x, 0, 0x20, 0x32, -1,
+                             OBJECT_D_LIFT, D_06000140);
+    }
+    if (!((this->dyna.actor.params >> 1) & 1)) {
+        func_80033480(globalCtx, &this->dyna.actor.world.pos, 120.0f, 12, 120, 100, 1);
+    } else if (((this->dyna.actor.params >> 1) & 1) == 1) {
+        func_80033480(globalCtx, &this->dyna.actor.world.pos, 60.0f, 8, 60, 100, 1);
+    }
+}
+
+void ObjLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLift* this = THIS;
+
+    ObjLift_InitDynaPoly(this, globalCtx, &D_060003D0, DPM_PLAYER);
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 2) & 0x3F)) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    Actor_SetScale(&this->dyna.actor, D_80B96A08[(this->dyna.actor.params >> 1) & 1]);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChainEntry);
+    this->unk168.x = Rand_ZeroOne() * 65535.5f;
+    this->unk168.y = Rand_ZeroOne() * 65535.5f;
+    this->unk168.z = Rand_ZeroOne() * 65535.5f;
+    func_80B9651C(this);
+    osSyncPrintf("(Dungeon Lift)(arg_data 0x%04x)\n", this->dyna.actor.params);
+}
+
+void ObjLift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLift* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
```

Comment:
we try to consistently add a newline after declared variables before the actual code starts. a small thing but it helps with readability